pytensor/tensor/subtensor.py

```diff
@@ -2561,7 +2561,11 @@ def c_code(self, node, name, input_names, output_names, sub):
             and y_.type.dtype not in complex_dtypes
         ):
             # Simple implementation for vector x, y cases
-            idx_may_be_neg = not (isinstance(idx_, Constant) and idx_.data.min() >= 0)
+            idx_may_be_neg = not (
+                # Empty idx needs no negative checks
+                idx_.type.shape[0] == 0
+                or (isinstance(idx_, Constant) and idx_.data.min() >= 0)
```

Comment:
we always have static shape for constants, so the check above precludes taking the min of an empty array, which was the original bug